### ISSUE TITLE
Sandia PV model blows up with solar zenith angle >= 89.9

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5528]
## Issue overview

Simple single zone model has PV and an inverter that is located in the zone fails with these errors:

```
   ** Severe  ** Temperature (low) out of bounds [-111879.06] for zone="ZONA FRACCAOA", for surface="PAREDE SUL1"
   **   ~~~   **  During Warmup & Sizing, Environment=INVERNO1, at Simulation time=01/02 08:07 - 08:08
   **   ~~~   ** Zone="ZONA FRACCAOA", Diagnostic Details:
   **   ~~~   ** ...Internal Heat Gain [1.239] W/m2
   **   ~~~   ** ...Infiltration/Ventilation [7.697E-002] m3/s
   **   ~~~   ** ...Mixing/Cross Mixing [0.000] m3/s
   **   ~~~   ** ...Zone is part of HVAC controlled system.
   ** Severe  ** Convergence error in SolveForWindowTemperatures for window VãO ENVIDRAçADO SUL (1)
   **   ~~~   **  During Warmup & Sizing, Environment=INVERNO1, at Simulation time=01/02 08:07 - 08:08
   **   ~~~   ** Glazing face index = 1 ; new temperature =NANC  ; previous temperature = -20379.2654C
   **   ~~~   ** Glazing face index = 2 ; new temperature =NANC  ; previous temperature = -21670.1720C
   **   ~~~   ** Glazing face index = 3 ; new temperature =NANC  ; previous temperature = -91506.3674C
   **   ~~~   ** Glazing face index = 4 ; new temperature =NANC  ; previous temperature = -92797.2859C
   **  Fatal  ** Program halted because of convergence error in SolveForWindowTemperatures for window VãO ENVIDRAçADO SUL (1)
```

Deleting the zone name from the inverter object (placing it outdoors) fixes the zone problem but PV generation is very large and very negative.  

The real problem is the function Photovoltaics::AbsoluteAirMass is getting called with a zenith angle of 
89.989 degrees.   Because of a div-by-zero trap as currently formulated this returns an air mass of 999.0 which when used in later correlations the values blow up and the PV is producing very large negative values.  (The inverter goes ahead and works with the negatives and zone internal gains are large and negative.)  
### Details

Some additional details for this issue (if relevant):
- Win64
- v8.4.0 and current develop https://github.com/NREL/EnergyPlus/commit/21ae200105e3831fab28c726b10e3de8c58e777e
- Helpdesk ticket 11149
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] [Defect file](https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/5000s/5528) added, run with PRT_Porto.085450_IWEC.epw (but that probably doesn't matter)
- [x] Ticket added to Pivotal for defect (development team task)
- [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by EnergyArchmage** [comment]
Note that load center internal gains issues have been addressed in PR 5512, which is not yet merged into develop.
- [x] Find value better than 999 for air mass to return when zenith angle >= 89.9 
- [x] Trap to protect against negative electric production results in load center and in PV generators

**Comment by EnergyArchmage** [comment]
https://github.com/NREL/EnergyPlus/pull/5529

**Comment by Myoldmopar** [comment]
Closed via #5529
